code/obj/machinery/photocopier.dm

```diff
@@ -29,246 +39,798 @@ TYPEINFO(/obj/machinery/photocopier)
+		P.sizey = src.print_info["sizey"]
+		P.scrollbar = src.print_info["scrollbar"]
+		P.overlays = src.print_info["overlays"]
+		return P
+
+	proc/create_paper_photo()
+		var/obj/item/paper/printout/P = new/obj/item/paper/printout(src)
+		P.desc = src.print_info[1]
+		P.print_icon = src.print_info[2]
+		P.print_icon_state = src.print_info[3]
+		return P
+
+	proc/create_wanted_poster()
+		var/obj/item/poster/titled_photo/W = new/obj/item/poster/titled_photo(src)
+		W.name = src.print_info["name"]
+		W.desc = src.print_info["desc"]
+		W.icon_state = src.print_info["icon_state"]
+		W.poster_image = src.print_info["poster_image"]
+		W.poster_image_old = src.print_info["poster_image_old"]
+		W.photo = src.print_info["photo"] = W.photo
+		W.line_title = src.print_info["line_title"]
+		W.poster_HTML = src.print_info["poster_HTML"]
+		W.line_photo_subtitle = src.print_info["line_photo_subtitle"]
+		W.line_below_photo = src.print_info["line_below_photo"]
+		W.line_b1 = src.print_info["line_b1"]
+		W.line_b2 = src.print_info["line_b2"]
+		W.line_b3 = src.print_info["line_b3"]
+		W.author = src.print_info["author"]
+		var/list/plist = src.print_info["plist"]
+		W.plist = plist.Copy()
+		return W
+
+	proc/load_stuff(var/obj/item/w, var/mob/user) // Load paper into the copier here
+		if (istype(w, /obj/item/paper))
+			if (istype(w, /obj/item/paper/book) || istype(w, /obj/item/paper/newspaper))
+				return;
+			if (src.paper_amount >= MAX_SHEETS)
+				boutput(user, SPAN_ALERT("You can't fit any more paper into \the [src]."))
+				effect_fail()
+				return
+			var/obj/item/paper/P = w
+			if (P.info != "" && tgui_alert(user, "This paper has writing on it, are you sure you want to put it in the inlet tray?", "Warning", list("Yes", "No")) == "No")
+				return
+			boutput(user, "You load the sheet of paper into \the [src].")
+			src.paper_amount++
+			qdel(w)
+
+		else if (istype(w, /obj/item/paper_bin))
+			var/obj/item/paper_bin/P = w
+			if (src.paper_amount >= MAX_SHEETS)
+				boutput(user, SPAN_ALERT("You can't fit any more paper into \the [src]."))
+			else if ((P.amount_left + src.paper_amount) >= MAX_SHEETS)
+				P.amount_left -= MAX_SHEETS - src.paper_amount
+				P.update()
+				src.paper_amount = MAX_SHEETS
+				boutput(user, SPAN_ALERT("You fill the paper tray in \the [src]."))
+			else
+				src.paper_amount += P.amount_left
+				P.amount_left = 0
+				P.update()
+				boutput(user, "You load the paper into \the [src].")
+
+		/*
+		else if (istype(w, /obj/item/card/id) && emagged)
+			var/obj/item/card/id/I = w
+			var/isBlank = TRUE
+			// isBlank = isBlank && I.name == "identification card"
+			isBlank = isBlank && I.icon_state == "id"
+			isBlank = isBlank && I.item_state == "card-id"
+			isBlank = isBlank && I.access.len == 0
+			isBlank = isBlank && I.registered == null
+			isBlank = isBlank && I.assignment == null
+			isBlank = isBlank && I.title == null
+			isBlank = isBlank && I.emagged == FALSE
+			isBlank = isBlank && I.money == 0 */
```

Comment:
Minor gripe but please remove this and the other ID cloning code instead of just commenting it, I know it sucks to remove something you put effort into but otherwise we end up with files filled with weird unused and unmaintained code everywhere.

Reply:
You're absolutely right. For the "isBlank" code in particular, should I move that to the ID card class as a new proc in case someone else wants to use it, or just get rid of it entirely since it probably wouldn't be maintained up until that point?

Reply:
Uhh I'd probably just remove it, I can't think of another reason you'd want to check that exact condition.

Reply:
Other use cases where we currently want to check an ID state are:
Agent card for their description (just checks if the name is still "Agent Card")
Emagging, of which there is a stat for.
Other use cases like this would probably just use their own method